### PR TITLE
closes #1193 : Issue 1193

### LINKED ISSUE #1193: Assess current destroy support
* Destroy a cache having a clustered tier
  * No other clients can be connected
  * Still closes the local cache
  * Throws to indicate did not destroy
* Destroy a cache manager having a clustered entity
  * Requires no clients have fetched the entity or hold the maintenance lease
  * Still destroys all local state!
  * Throws on failure, CM back to un-initialized state

Need to know what remains to be done and how we handle corner cases

### PR DESCRIPTION
**Pull request by AbfrmBlr**
Added some tests to verify destroy behavior for cache and cachemanager for user facing apis

### CODE REVIEW

**Comment by ljacomet** [comment]
The `Closes` or other word to trigger issue closing is better put as a commit message, the second in this case.
Also could you fix the typos in the messages? destory -> destroy

File: clustered/client/src/test/java/org/ehcache/clustered/client/CacheManagerDestroyTest.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  @Test
+  public void testDestroyCacheManagerWithMultipleClients() throws CachePersistenceException {
+    PersistentCacheManager persistentCacheManager1 = clusteredCacheManagerBuilder.build(true);
+    PersistentCacheManager persistentCacheManager2 = clusteredCacheManagerBuilder.build(true);
+
+    persistentCacheManager1.close();
+
+    try {
+      persistentCacheManager1.destroy();
+    } catch (StateTransitionException e) {
```
Comment (albinsuresh) [inline clustered/client/src/test/java/org/ehcache/clustered/client/CacheManagerDestroyTest.java]: Wouldn't it be safer to add an Assert.fail here after line 81?

File: clustered/client/src/test/java/org/ehcache/clustered/client/CacheManagerDestroyTest.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        .withCache("test", CacheConfigurationBuilder.newCacheConfigurationBuilder(Long.class, String.class,
+            ResourcePoolsBuilder.newResourcePoolsBuilder()
+                .with(ClusteredResourcePoolBuilder.fixed("primary-server-resource", 32, MemoryUnit.MB))));
+
+    PersistentCacheManager persistentCacheManager1 = cacheManagerBuilder.build(true);
+    PersistentCacheManager persistentCacheManager2 = cacheManagerBuilder.build(true);
+
+    persistentCacheManager1.close();
+    try {
+      persistentCacheManager1.destroy();
```
Comment (albinsuresh) [inline clustered/client/src/test/java/org/ehcache/clustered/client/CacheManagerDestroyTest.java]: Same as above.
